src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingContainer.java

```diff
@@ -194,20 +191,25 @@ else if (block instanceof BlockMinecoloniesRack)
      * @param tagName the name of the tag to query
      * @return the BlockPos, or null if not found
      */
-    protected BlockPos getFirstLocationFromTag(String tagName)
+    @Nullable
+    protected BlockPos getFirstLocationFromTag(@NotNull final String tagName)
     {
-        if (tileEntity != null && !tileEntity.getPositionedTags().isEmpty())
+        return getLocationsFromTag(tagName).stream().findFirst().orElse(null);
```

Comment:
stream, findFirst, seems like isEmpty ? null : get(0)

Reply:
That would be equivalent, but not a one-liner.

Reply:
Yeah, but performs much better because it doesn't require to create a stream

Reply:
k